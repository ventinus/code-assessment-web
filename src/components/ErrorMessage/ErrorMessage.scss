@import '../../scss/variables/mixins.scss';
@import '../../scss/variables/variables.scss';

.error-message {
  background-color: $red-2;
  padding: 10px 20px;
}
