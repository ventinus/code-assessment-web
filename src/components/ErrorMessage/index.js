import React from 'react'
import PropTypes from 'prop-types'

import './ErrorMessage.css'

const ErrorMessage = ({ style = {}, children }) => {
  if (!children) return null

  return (
    <div className="error-message" style={style}>
      <p>There was an issue completing checkout: {children}</p>
    </div>
  )
}

ErrorMessage.propTypes = {
  style: PropTypes.object,
  children: PropTypes.any,
}

export default ErrorMessage
