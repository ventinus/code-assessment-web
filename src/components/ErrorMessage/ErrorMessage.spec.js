import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import ErrorMessage from '.'

describe('ErrorMessage component', () => {
  it('renders correctly with an error message', () => {
    const tree = renderer
      .create(<ErrorMessage>There was an error</ErrorMessage>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('does not render without children present', () => {
    const component = shallow(<ErrorMessage></ErrorMessage>)
    expect(component.get(0)).toBe(null)
  })
})
