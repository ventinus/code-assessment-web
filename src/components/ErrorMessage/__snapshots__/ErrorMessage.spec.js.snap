// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component renders correctly with an error message 1`] = `
<div
  className="error-message"
  style={Object {}}
>
  <p>
    There was an issue completing checkout: 
    There was an error
  </p>
</div>
`;
